services/headless-lms/chatbot/src/chatbot_tools/course_progress.rs

```diff
@@ -0,0 +1,650 @@
+            order_number: 1,
+            progress: UserCourseProgress {
+                course_module_id: Uuid::nil(),
+                course_module_name: "Example base module".to_string(),
+                course_module_order_number: 1,
+                score_given: 0.0,
+                score_required: Some(9),
+                score_maximum: Some(10),
+                total_exercises: Some(10),
+                attempted_exercises: None,
+                attempted_exercises_required: None,
+            },
+            automatic_completion: true,
+            requires_exam: false,
+        }];
+        let tool = CourseProgressTool::new_mock("Advanced Chatbot Course".to_string(), progress);
+        let output = tool.get_tool_output();
+
+        let expected_output =
+"Result: [output]The user is completing a course called Advanced Chatbot Course. Their progress on this course is the following: On this course, there are available a total of 10 exercises and 10 exercise points. To pass this course, it's required to gain 9 exercise points. The user has gained 0.0 points. To pass this course, the user needs to gain 9.0 more points.\n[/output]
+
+Instructions for describing the output: [instructions]Describe this information in a short, clear way with no or minimal bullet points. Only give information that is relevant to the user's question. If the course has multiple modules and the user asks something like 'how to pass the course', by default describe the passing criteria and requirements of the base module. Encourage the user to ask further questions about other modules if needed.[/instructions]".to_string();
+
+        assert_eq!(output, expected_output);
+    }
+
+    #[test]
+    fn test_course_progress_output_pts_required_none() {
+        let progress = vec![CourseProgressInfo {
+            order_number: 1,
+            progress: UserCourseProgress {
+                course_module_id: Uuid::nil(),
+                course_module_name: "Example base module".to_string(),
+                course_module_order_number: 1,
+                score_given: 0.0,
+                score_required: None,
+                score_maximum: Some(10),
+                total_exercises: Some(10),
+                attempted_exercises: None,
+                attempted_exercises_required: Some(10),
+            },
+            automatic_completion: true,
+            requires_exam: false,
+        }];
+        let tool = CourseProgressTool::new_mock("Advanced Chatbot Course".to_string(), progress);
+        let output = tool.get_tool_output();
+
+        let expected_output =
+"Result: [output]The user is completing a course called Advanced Chatbot Course. Their progress on this course is the following: On this course, there are available a total of 10 exercises and 10 exercise points. To pass this course, it's required to attempt 10 exercises. The user has not attempted any exercises. To pass this course, they need to attempt 10 more exercises.\n[/output]
```

Comment:
Even though only attempting exercises is required to pass the course, we could still tell how many points the user has gotten, in case the student asks about the points.